cypress-io/cypress
# chore: adding support for cross-origin

### Additional details

All existing cypress events were supported for cross-origin except for `viewport:changed`.
* Updated the event listener to also listen for `viewport:changed` on the `primaryOriginCommunicator`
* Updated the `preprocessSnapshotForSerialization` function to just return the existing snapshot when the protocol is enabled since the snapshot is already serializable.

### Steps to test

* Checkout `mschile/protocol/cross-origin` from the services repo
* Record a test that used `cy.origin`
* Verify the database is correct

### How has the user experience changed?


### PR Tasks


- [n/a] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
